ParquetSharp.DataFrame/DataFrameExtensions.cs

```diff
@@ -8,58 +8,100 @@ namespace ParquetSharp
     public static class DataFrameExtensions
     {
         /// <summary>
-        /// Write a DataFrame in Parquet format
+        /// Writes a Dataframe in Parquet format using existing writer.
         /// </summary>
-        /// <param name="dataFrame">The DataFrame to write</param>
+        /// <param name="dataFrame">DataFrame to write.</param>
+        /// <param name="fileWriter">Writer to use.</param>
+        /// <param name="rowGroupSize">Maximum number of rows per row group</param>
+        public static void ToParquet(this DataFrame dataFrame, ParquetFileWriter fileWriter, int rowGroupSize = 1024 * 1024)
+        {
+            long numRows = dataFrame.Rows.Count;
+            long offset = 0L;
+
+            while (offset < numRows)
+            {
+                int batchSize = (int)Math.Min(numRows - offset, rowGroupSize);
+                using RowGroupWriter rowGroupWriter = fileWriter.AppendRowGroup();
+                foreach (DataFrameColumn dataFrameColumn in dataFrame.Columns)
+                {
+                    using ColumnWriter columnWriter = rowGroupWriter.NextColumn();
+                    using LogicalColumnWriter logicalWriter = columnWriter.LogicalWriter();
+                    logicalWriter.Apply(new DataFrameWriter(dataFrameColumn, offset, batchSize));
+                }
+
+                offset += batchSize;
+            }
+        }
+
+        /// <summary>
+        /// Write DataFrames in Parquet format
+        /// </summary>
+        /// <param name="dataFrame">DataFrames to write</param>
         /// <param name="path">Path to write to</param>
         /// <param name="writerProperties">Optional writer properties that override the default properties</param>
         /// <param name="logicalTypeOverrides">Mapping from column names to Parquet logical types,
         /// overriding the default logical types. When writing decimal columns, a logical type must be provided
         /// to specify the precision and scale to use.</param>
         /// <param name="rowGroupSize">Maximum number of rows per row group</param>
         public static void ToParquet(
-            this DataFrame dataFrame, string path, WriterProperties? writerProperties = null,
+            this IEnumerable<DataFrame> dataFrames, string path, WriterProperties? writerProperties = null,
```

Comment:
easy overload to write multiple dataframes. They must be of the same schema. this is currenlty not enforced.
TODO: ensure that schema is same in all dataframes in a sequence.